### PR TITLE
K8SPSMDB-258 MongoDB 4.4 version integration

### PR DESCRIPTION
**Pull request by cap1984**
[K8SPSMDB-258](https://jira.percona.com/browse/K8SPSMDB-258)

A few changes for newcoming db version

### CODE REVIEW

**Review by tplavcic — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | failed |
| upgrade | passed |
| storage | passed |
| self-healing | passed |
| monitoring | failed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| demand-backup | passed |
| monitoring-2-0 | failed |
| operator-self-healing | passed |
| security-context | passed |
| scheduled-backup | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-441-158f8e11

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| self-healing | passed |
| monitoring | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

File: e2e-tests/upgrade/run [hunk 1]
```diff
@@ -28,6 +28,11 @@ IMAGE_BACKUP='percona/percona-server-mongodb-operator:1.1.0-backup'
 
 if [[ "${MONGO_VER}" == "4.2" ]]; then
     IMAGE_MONGOD="percona/percona-server-mongodb-operator:1.1.0-mongod4.0"
+elif [[ "${MONGO_VER}" == "4.4" ]]; then
+    API='psmdb.percona.com/v1-4-0'
+    IMAGE='percona/percona-server-mongodb-operator:1.4.0'
+    IMAGE_MONGOD="percona/percona-server-mongodb-operator:1.4.0-mongod4.2"
+    IMAGE_BACKUP='percona/percona-server-mongodb-operator:1.4.0-backup'
```
Comment (delgod) [inline e2e-tests/upgrade/run:35]: let us test all versions since 1.4.0
I prefer to avoid adding `statefulset_upgrade-rs0-4.4.yml`

**Review by delgod — DISMISSED** [review]

File: e2e-tests/upgrade/run (hunk 1 above)
Comment (cap1984) [reply e2e-tests/upgrade/run]: done

**Review by tplavcic — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| scheduled-backup | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring | passed |
| monitoring-2-0 | failed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-441-57e15f43

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring | passed |
| storage | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-441-893f5045

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

File: e2e-tests/upgrade/run [hunk 2]
```diff
@@ -20,15 +20,11 @@ if [[ "${IMAGE_MONGOD}" == *"percona-server-mongodb-operator"* ]]; then
...
-IMAGE_MONGOD="percona/percona-server-mongodb-operator:1.1.0-mongod$MONGO_VER"
-IMAGE_BACKUP='percona/percona-server-mongodb-operator:1.1.0-backup'
-
-if [[ "${MONGO_VER}" == "4.2" ]]; then
-    IMAGE_MONGOD="percona/percona-server-mongodb-operator:1.1.0-mongod4.0"
-fi
+API='psmdb.percona.com/v1-4-0'
+IMAGE_PMM='percona/percona-server-mongodb-operator:1.4.0-pmm'
+IMAGE='percona/percona-server-mongodb-operator:1.4.0'
+IMAGE_MONGOD="percona/percona-server-mongodb-operator:1.4.0-mongod4.2"
```
Comment (delgod) [inline e2e-tests/upgrade/run:26]:
```suggestion
IMAGE_MONGOD="percona/percona-server-mongodb-operator:1.4.0-mongod$MONGO_VER"
```

**Review by delgod — CHANGES_REQUESTED** [review]

File: e2e-tests/upgrade/run (hunk 2 above)
Comment (cap1984) [reply e2e-tests/upgrade/run]: done

**Review by delgod — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| arbiter | passed |
| scheduled-backup | passed |
| service-per-pod | failed |
| limits | passed |
| scaling | passed |
| security-context | passed |

**Review by tplavcic — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| self-healing | passed |
| security-context | passed |
| arbiter | passed |
| operator-self-healing | passed |
| service-per-pod | failed |
| scheduled-backup | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring | passed |
| monitoring-2-0 | failed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| arbiter | passed |
| service-per-pod | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-441-41115647

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

**Pull request merged**